src/main/java/com/group6/harmoniq/controllers/SpotifyController.java

```diff
@@ -0,0 +1,159 @@
+package com.group6.harmoniq.controllers;
+
+import java.security.SecureRandom;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.CookieValue;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.ui.Model;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletResponse;
+
+import io.github.cdimascio.dotenv.Dotenv;
+
+
+
+@Controller
+public class SpotifyController {
+
+    String client_id, client_secret, redirect_uri;
+
+    public SpotifyController() {
+    if("development".equals(System.getenv("ENVIRONMENT")))
+{
+        client_id = System.getenv("SPOTIFY_CLIENT_ID");
```

Comment:
But we can still use the `.env` locally as well, right?! Wouldn't the `Dotenv` library work locally too?

Reply:
Yes the Dotenv library works locally and uses the .env file, but when we deploy to render, the .env file isn't on our github repo so it doesn't work, and so we have to input the environment variables manually on render.com when creating the website. So if we're in a development environment, i.e. render, we use System.getenv("SPOTIFY_CLIENT_ID") to get those environment variables we input into render, but if we're in a local environment, then we use client_id = dotenv.get("SPOTIFY_CLIENT_ID"); ,which gets that value from the .env file.

Reply:
I see. Thanks for clarifying!